admin-ui/src/component/my-units/create-reservation/validator.ts

```diff
@@ -1,62 +1,152 @@
-import coreJoi from "joi";
-import joiDate from "@joi/date";
-import { startOfDay } from "date-fns";
+import { z } from "zod";
+import { subDays } from "date-fns";
 import { ReservationUnitsReservationUnitReservationStartIntervalChoices } from "common/types/gql-types";
-import { getDayIntervals } from "common/src/calendar/util";
-import i18n from "../../../i18n";
-import { ReservationType } from "./types";
-
-const joi = coreJoi.extend(joiDate) as typeof coreJoi;
-
-const TIME_PATTERN = /^[0-9+]{2}:[0-9+]{2}$/;
-
-const timeWithIntervals = (
-  interval: ReservationUnitsReservationUnitReservationStartIntervalChoices = ReservationUnitsReservationUnitReservationStartIntervalChoices.Interval_15Mins
-): coreJoi.StringSchema => {
-  const intervals = getDayIntervals("00:00", "23:59", interval).map(
-    (startHMS) => startHMS.substring(0, 5)
-  );
-  return joi
-    .string()
-    .pattern(TIME_PATTERN)
-    .valid(...intervals)
-    .messages({
-      "any.only": i18n.t("ReservationDialog.validation.interval", {
-        interval: `${intervals.join(", ").substring(0, 50)}...`,
-      }),
+import { intervalToNumber } from "./utils";
+
+const THREE_YEARS_MS = 3 * 365 * 24 * 60 * 60 * 1000;
+const TIME_PATTERN = /^[0-2][0-9]:[0-5][0-9]$/;
+
+export const ReservationTypes = ["STAFF", "BEHALF", "BLOCKED"] as const;
```

Comment:
The values could be used to validate the values or maybe request api to return enum to represent the values.